Add in `github_actions_organization_secret`

jacobfoard opened the pull request:
This fixes #468. 

This almost exclusively follows the same patterns as `github_actions_secret`, but happy to address any feedback!

I also fixed the `updated_at` and `created_at` fields on `github_actions_secret` from the string `"default"` to being a proper timestamp.

jcudit commented:
🎉 looking good on a quick glance. 

Can we get a new file added under the website directory that documents this new resource?

I believe there would also be a change to add the new documentation to the side navigation area. I can link an example when at my keyboard.

Aiming to come back with a more thorough review as well. Thanks for kicking this one off @jacobfoard !

jacobfoard commented:
I added in handling a couple edge cases I hadn't previously tested, added/updated docs, and needed to add in the `repo_id` for use with the `selected` visibility.

jcudit commented:
@jacobfoard gave this another read and caught a typo.  `github_actions_orginization_secret` should be `github_actions_organization_secret`.  Can you replace all occurrences when you get a moment?

jacobfoard commented:
@jcudit Done, and commits squashed as well.

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,117 @@
+		updatedSecretValue := "updated_super_secret_value"
+
+		config := fmt.Sprintf(`
+			resource "github_actions_organization_secret" "test_secret" {
+			  secret_name      = "test_secret_name"
+			  plaintext_value  = "%s"
+			}
```
[Tests seems to be failing](https://github.com/terraform-providers/terraform-provider-github/actions/runs/361785493) due to a missing parameter here:

```
2020-11-13T14:56:31.9129101Z --- FAIL: TestAccGithubActionsOrganizationSecret (0.03s)
2020-11-13T14:56:31.9130511Z     --- FAIL: TestAccGithubActionsOrganizationSecret/creates_and_updates_secrets_without_error (0.02s)
2020-11-13T14:56:31.9132146Z         --- SKIP: TestAccGithubActionsOrganizationSecret/creates_and_updates_secrets_without_error/with_an_anonymous_account (0.00s)
2020-11-13T14:56:31.9133346Z             resource_github_actions_organization_secret_test.go:70: anonymous account not supported for this operation
2020-11-13T14:56:31.9134760Z         --- SKIP: TestAccGithubActionsOrganizationSecret/creates_and_updates_secrets_without_error/with_an_individual_account (0.00s)
2020-11-13T14:56:31.9135969Z             resource_github_actions_organization_secret_test.go:74: individual account not supported for this operation
2020-11-13T14:56:31.9137406Z         --- FAIL: TestAccGithubActionsOrganizationSecret/creates_and_updates_secrets_without_error/with_an_organization_account (0.02s)
2020-11-13T14:56:31.9138701Z             testing.go:654: Step 0 error: config is invalid: Missing required argument: The argument "visibility" is required, but no definition was found.
2020-11-13T14:56:31.9140034Z     --- FAIL: TestAccGithubActionsOrganizationSecret/deletes_secrets_without_error (0.01s)
2020-11-13T14:56:31.9142351Z         --- SKIP: TestAccGithubActionsOrganizationSecret/deletes_secrets_without_error/with_an_anonymous_account (0.00s)
2020-11-13T14:56:31.9143617Z             resource_github_actions_organization_secret_test.go:106: anonymous account not supported for this operation
2020-11-13T14:56:31.9145133Z         --- SKIP: TestAccGithubActionsOrganizationSecret/deletes_secrets_without_error/with_an_individual_account (0.00s)
2020-11-13T14:56:31.9146407Z             resource_github_actions_organization_secret_test.go:110: individual account not supported for this operation
2020-11-13T14:56:31.9147905Z         --- FAIL: TestAccGithubActionsOrganizationSecret/deletes_secrets_without_error/with_an_organization_account (0.01s)
2020-11-13T14:56:31.9149270Z             testing.go:654: Step 0 error: config is invalid: Missing required argument: The argument "visibility" is required, but no definition was found.
2020-11-13T14:56:31.9149955Z FAIL
2020-11-13T14:56:31.9150779Z FAIL	github.com/terraform-providers/terraform-provider-github/github	0.041s
2020-11-13T14:56:31.9151544Z FAIL
```

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,117 @@
+		secretValue := "super_secret_value"
+
+		config := fmt.Sprintf(`
+				resource "github_actions_organization_secret" "test_secret" {
+					secret_name      = "test_secret_name"
+					plaintext_value  = "%s"
```
Same as above ☝️

jcudit reviewed (commented):
Spotted a few more fixes to make 😄 .

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,119 @@
+		checks := map[string]resource.TestCheckFunc{
+			"before": resource.ComposeTestCheckFunc(
+				resource.TestCheckResourceAttr(
+					"github_actions_secret.test_secret", "plaintext_value",
```
```suggestion
					"github_actions_organization_secret.test_secret", "plaintext_value",
```

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,119 @@
+					secretValue,
+				),
+				resource.TestCheckResourceAttrSet(
+					"github_actions_secret.test_secret", "created_at",
```
```suggestion
					"github_actions_organization_secret.test_secret", "created_at",
```

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,119 @@
+					"github_actions_secret.test_secret", "created_at",
+				),
+				resource.TestCheckResourceAttrSet(
+					"github_actions_secret.test_secret", "updated_at",
```
```suggestion
					"github_actions_organization_secret.test_secret", "updated_at",
```

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,119 @@
+			),
+			"after": resource.ComposeTestCheckFunc(
+				resource.TestCheckResourceAttr(
+					"github_actions_secret.test_secret", "plaintext_value",
```
```suggestion
					"github_actions_organization_secret.test_secret", "plaintext_value",
```

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,119 @@
+					updatedSecretValue,
+				),
+				resource.TestCheckResourceAttrSet(
+					"github_actions_secret.test_secret", "created_at",
```
```suggestion
					"github_actions_organization_secret.test_secret", "created_at",
```

jcudit commented on github/resource_github_actions_organization_secret_test.go:
```diff
@@ -0,0 +1,119 @@
+					"github_actions_secret.test_secret", "created_at",
+				),
+				resource.TestCheckResourceAttrSet(
+					"github_actions_secret.test_secret", "updated_at",
```
```suggestion
					"github_actions_organization_secret.test_secret", "updated_at",
```

jcudit reviewed (commented):
Very close 🎉 .  Take a look at the test fixes and commit if all is 👌 .  Will look to release this in the upcoming week.

jacobfoard commented:
Alright, I believe the last of the copy pasta has been cleaned up. Thanks for your patience, I'll make sure my next contribution isn't as rushed 😅.

jcudit commented:
❤️  this contribution.  Looks like a popular one from the community's view, so 🙇  for making everyone's life a bit easier.  Aiming to release this week.
